resolve no-var violations in javascripts/projectsService.js

We have some very old ESLint rules that I'd like to clean up, and our use of `var` is one of them. ES6 introduces `let` and `const` as a way to make clearer when a variable can be used, and I'd like to reduce our use of `var` where we don't need it:

If you know a bit about JS you can help out by following these steps:

 - create a new branch from `gh-pages`
 - in `javascripts/projectsService.js` add this comment at the top of the file, before any source code:

```js
/* eslint no-var: [ "error" ] */
```

 - in a terminal, run `npm run lint`, which should emit a bunch of violations:

```
$ npm run lint

> up-for-grabs.net@1.0.0 lint /Users/shiftkey/src/up-for-grabs.net
> eslint javascripts/*.js tests/**/*.js


/Users/shiftkey/src/up-for-grabs.net/javascripts/projectsService.js
    8:3  error  Unexpected var, use let or const instead  no-var
   12:3  error  Unexpected var, use let or const instead  no-var
   ...
```

 - run `npm run lint-fix` to fix whatever can be automatically tidied up
 - review and address the remaining violations manually
 - run `npm run prettier-fix` to ensure the formatting of the changes is fine
 - [test the site locally](https://github.com/up-for-grabs/up-for-grabs.net#testing-the-site-locally)
 - commit the changes and open a PR
 - test out the Netlify deploy site (click the `Details` link) and confirm the client-side functionality is still working

<img width="709" src="https://user-images.githubusercontent.com/359239/66651591-c70eaf80-ec09-11e9-85bc-a8c58d81e2be.png">

--- a/javascripts/projectsService.js
+++ b/javascripts/projectsService.js
@@ -5,16 +5,18 @@
 /* eslint function-paren-newline: [ "off" ] */
 /* eslint implicit-arrow-linebreak: [ "off" ] */
 /* eslint no-confusing-arrow: [ "off" ] */
+/* eslint no-var: [ "error" ] */
 
 // @ts-nocheck
 
 // required for loading into a NodeJS context
 if (typeof define !== 'function') {
+  /* eslint-disable-next-line no-var */
   var define = require('amdefine')(module);
 }
 
 define(['underscore'], _ => {
-  var applyTagsFilter = function(projects, tagsArray, tags) {
+  const applyTagsFilter = function(projects, tagsArray, tags) {
     if (typeof tags === 'string') {
       tags = tags.split(',');
     }
@@ -25,7 +27,7 @@ define(['underscore'], _ => {
       return projects;
     }
 
-    var projectNames = _.uniq(
+    const projectNames = _.uniq(
       _.flatten(
         _.map(tags, tag => {
           // NOTE
@@ -60,7 +62,7 @@ define(['underscore'], _ => {
    *              projects in a sorted order
    * @param Array names : This is an array with the given name filters.
    */
-  var applyNamesFilter = function(projects, projectNamesSorted, names) {
+  const applyNamesFilter = function(projects, projectNamesSorted, names) {
     if (typeof names === 'string') {
       names = names.split(',');
     }
@@ -92,7 +94,7 @@ define(['underscore'], _ => {
    *        labels in a sorted order
    * @param Array labels : This is an array with the given label filters.
    */
-  var applyLabelsFilter = function(projects, projectLabelsSorted, labels) {
+  const applyLabelsFilter = function(projects, projectLabelsSorted, labels) {
     label_indices = labels;
 
     if (typeof labels === 'string') {
@@ -132,12 +134,12 @@ define(['underscore'], _ => {
     return _.flatten(results, (arr1, arr2) => arr1.append(arr2));
   };
 
-  var TagBuilder = function() {
-    var _tagsMap = {},
-      _orderedTagsMap = null;
+  const TagBuilder = function() {
+    const _tagsMap = {};
+    let _orderedTagsMap = null;
 
     this.addTag = function(tag, projectName) {
-      var tagLowerCase = tag.toLowerCase();
+      const tagLowerCase = tag.toLowerCase();
       if (!_.has(_tagsMap, tagLowerCase)) {
         _tagsMap[tagLowerCase] = {
           name: tag,
@@ -145,7 +147,7 @@ define(['underscore'], _ => {
           projects: [],
         };
       }
-      var _entry = _tagsMap[tagLowerCase];
+      const _entry = _tagsMap[tagLowerCase];
       _entry.frequency++;
       _entry.projects.push(projectName);
     };
@@ -162,8 +164,8 @@ define(['underscore'], _ => {
     };
   };
 
-  var extractTags = function(projectsData) {
-    var tagBuilder = new TagBuilder();
+  const extractTags = function(projectsData) {
+    const tagBuilder = new TagBuilder();
     _.each(projectsData, entry => {
       _.each(entry.tags, tag => {
         tagBuilder.addTag(tag, entry.name);
@@ -172,25 +174,25 @@ define(['underscore'], _ => {
     return tagBuilder.getTagsMap();
   };
 
-  var extractProjectsAndTags = function(projectsData) {
+  const extractProjectsAndTags = function(projectsData) {
     return {
       projects: projectsData,
       tags: extractTags(projectsData),
     };
   };
 
-  var ProjectsService = function(projectsData) {
-    var _projectsData = extractProjectsAndTags(projectsData);
-    var tagsMap = {};
-    var namesMap = {};
-    var labelsMap = {};
+  const ProjectsService = function(projectsData) {
+    const _projectsData = extractProjectsAndTags(projectsData);
+    const tagsMap = {};
+    const namesMap = {};
+    const labelsMap = {};
 
-    var canStoreOrdering =
+    const canStoreOrdering =
       JSON &&
       sessionStorage &&
       sessionStorage.getItem &&
       sessionStorage.setItem;
-    var ordering = null;
+    let ordering = null;
     if (canStoreOrdering) {
       ordering = sessionStorage.getItem('projectOrder');
       if (ordering) {
@@ -210,9 +212,9 @@ define(['underscore'], _ => {
       }
     }
 
-    var all_projects = _.map(ordering, i => _projectsData.projects[i]);
+    const all_projects = _.map(ordering, i => _projectsData.projects[i]);
 
-    var projects = _.filter(all_projects, project =>
+    const projects = _.filter(all_projects, project =>
       project.stats ? project.stats['issue-count'] > 0 : true
     );
 
@@ -231,7 +233,7 @@ define(['underscore'], _ => {
     });
 
     this.get = function(tags, names, labels) {
-      var filtered_projects = projects;
+      let filtered_projects = projects;
       if (names && names.length) {
         filtered_projects = applyNamesFilter(
           filtered_projects,

